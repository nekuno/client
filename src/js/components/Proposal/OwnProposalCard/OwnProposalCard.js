import PropTypes from 'prop-types';
import React, { Component } from 'react';
import RoundedImage from '../../ui/RoundedImage/RoundedImage.js';
import RoundedIcon from '../../ui/RoundedIcon/RoundedIcon.js';
import styles from './OwnProposalCard.scss';
import translate from '../../../i18n/Translate';

@translate('OwnProposalCard')
export default class OwnProposalCard extends Component {

    static propTypes = {
        image         : PropTypes.string.isRequired,
        title         : PropTypes.string.isRequired,
        description   : PropTypes.string.isRequired,
        type          : PropTypes.string.isRequired,
        photos        : PropTypes.array,
        onClickHandler: PropTypes.func
    };

    handleClick() {
        if (this.props.onClickHandler) {
            this.props.onClickHandler();
        }
    }

    render() {
        const {image, title, description, type, photos, strings} = this.props;
        let icon = null;
        let background = null;
        switch (type) {
            case 'work':
                icon = 'paperclip';
                background = '#63CAFF';
                break;
            case 'leisure-plan':
                icon = 'send';
                background = '#D380D3';
                break;
            case 'experience-plan':
                icon = 'compass';
                background = '#7BD47E';
                break;
        }

        return (
            <div className={styles.ownProposalCard} onClick={this.handleClick.bind(this)}>
                <div className={styles.frame}>
                    <div className={styles.topData}>
                        <h2>{title}</h2>
                    </div>
                    {icon ?
                        <div className={styles.type}>
                            <RoundedIcon icon={icon} size={'medium'} background={background} fontSize={'24px'}/>
                        </div>
                        : null
                    }
                    <div className={styles.proposalImage}>
                        <img src={image}/>
                    </div>
                    <div className={styles.description}>
                        <div className={styles.descriptionText + ' ' + styles.truncate}>{description}</div>
                    </div>
                    {photos &&
                    <div className={styles.userData}>
                        <div className={styles.photos}>
                            {photos.map((photo, index) =>
                                <div className={styles.photo} key={index}>
                                    <RoundedImage size={'x-small'} url={photo}/>
                                </div>
                            )}
                        </div>
                        <div className={styles.matches}>
                            {photos.length} {strings.matches}
                        </div>
                    </div>
                    }

                </div>
            </div>
        );
    }
}

OwnProposalCard.defaultProps = {
    strings: {
        matches: 'Matches',
    }
};