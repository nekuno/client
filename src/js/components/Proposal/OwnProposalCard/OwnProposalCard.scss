@import '../../../../scss/_partials/colors';

.own-proposal-card {
  cursor: pointer;
  display: inline-block;
  font-weight: 600;
  margin: auto;
  position: relative;
  text-align: left;
  width: 80%;

  .frame {
    background: #FFF;
    border-radius: 10px;
    -moz-border-radius: 10px;
    -webkit-border-radius: 10px;
    box-shadow: 0 0 5px $gray-border;
    color: #6C6F82;
    font-size: 14px;
    margin: 2.5%;
    padding: 0;
    width: 95%;

    .proposal-image {
      border-radius: 10px 10px 0 0;
      -moz-border-radius: 10px 10px 0 0;
      -webkit-border-radius: 10px 10px 0 0;
      height: 130px;
      overflow: hidden;
      text-align: center;
      width: 100%;

      img {
        border-radius: 10px 10px 0 0;
        -moz-border-radius: 10px 10px 0 0;
        -webkit-border-radius: 10px 10px 0 0;
        width: 100%;
      }

    }
    .top-data {
      color: white;
      height: 130px;
      padding: 0 5%;
      position: absolute;
      width: 80%;

      h2 {
        bottom: -10px;
        font-weight: 600;
        max-height: 75px;
        overflow: hidden;
        position: absolute;
      }
    }
    .description {
      padding: 16px;
      .description-text {
        font-weight: normal;
        height: 66px;
        overflow: hidden;
        // margin-top: 16px;
      }
    }
    .type {
      position: absolute;
      right: 16px;
      top: 16px;
    }
    .user-data {
      height: 60px;
      margin-top: 16px;
      overflow: hidden;
      padding: 0 5%;

      .photos {
        display: inline-block;
        margin-left: 6px;
        vertical-align: super;

        .photo {
          display: inline-block;
          margin-right: -30px;
        }
      }
      .matches {
        color: $gray-main-text;
        display: inline-block;
        line-height: 50px;
        height: 50px;
        margin-left: 35px;
        vertical-align: top;
      }
    }
  }
}