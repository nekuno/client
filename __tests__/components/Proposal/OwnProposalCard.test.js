import React from 'react';
import OwnProposalCard from '../../../src/js/components/Proposal/OwnProposalCard/OwnProposalCard.js';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test OwnProposalCard component', () => {
    const mockFn = jest.fn();
    const ownProposalCard = shallow(<OwnProposalCard title={'Lorem ipsum dolor sit amet'}
                                               image={'http://via.placeholder.com/360x180'}
                                               type={'professional-project'}
                                               photos={['http://via.placeholder.com/100x100/ff0000', 'http://via.placeholder.com/100x100/00ff00', 'http://via.placeholder.com/100x100/0000ff', 'http://via.placeholder.com/100x100/ff0000', 'http://via.placeholder.com/100x100/00ff00']}
                                               nickname={'JohnDoe'}
                                               resume={'Lorem ipsum dolor sit amet'}
                                               onClickHandler={mockFn}/>
    );

    it('should be defined', () => {
        expect(ownProposalCard).toBeDefined();
    });
    it('should render correctly', () => {
        expect(ownProposalCard).toMatchSnapshot();
    });
});
