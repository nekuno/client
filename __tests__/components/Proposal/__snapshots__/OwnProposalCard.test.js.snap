// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test OwnProposalCard component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TranslationComponent
    image="http://via.placeholder.com/360x180"
    nickname="JohnDoe"
    onClickHandler={[MockFunction]}
    photos={
      Array [
        "http://via.placeholder.com/100x100/ff0000",
        "http://via.placeholder.com/100x100/00ff00",
        "http://via.placeholder.com/100x100/0000ff",
        "http://via.placeholder.com/100x100/ff0000",
        "http://via.placeholder.com/100x100/00ff00",
      ]
    }
    resume="Lorem ipsum dolor sit amet"
    strings={
      Object {
        "matches": "Matches",
      }
    }
    title="Lorem ipsum dolor sit amet"
    type="professional-project"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "image": "http://via.placeholder.com/360x180",
      "locale": undefined,
      "nickname": "JohnDoe",
      "onClickHandler": [MockFunction],
      "photos": Array [
        "http://via.placeholder.com/100x100/ff0000",
        "http://via.placeholder.com/100x100/00ff00",
        "http://via.placeholder.com/100x100/0000ff",
        "http://via.placeholder.com/100x100/ff0000",
        "http://via.placeholder.com/100x100/00ff00",
      ],
      "resume": "Lorem ipsum dolor sit amet",
      "strings": Object {
        "matches": "Matches",
      },
      "title": "Lorem ipsum dolor sit amet",
      "type": "professional-project",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "image": "http://via.placeholder.com/360x180",
        "locale": undefined,
        "nickname": "JohnDoe",
        "onClickHandler": [MockFunction],
        "photos": Array [
          "http://via.placeholder.com/100x100/ff0000",
          "http://via.placeholder.com/100x100/00ff00",
          "http://via.placeholder.com/100x100/0000ff",
          "http://via.placeholder.com/100x100/ff0000",
          "http://via.placeholder.com/100x100/00ff00",
        ],
        "resume": "Lorem ipsum dolor sit amet",
        "strings": Object {
          "matches": "Matches",
        },
        "title": "Lorem ipsum dolor sit amet",
        "type": "professional-project",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
